src/smc_sagews/smc_sagews/test_without_server/conftest.py

```diff
@@ -0,0 +1,88 @@
+def pytest_addoption(parser):
+    """specify comma-delimited list of kernel names to limit test"""
+    parser.addoption("--kname", action="store", help="kernel name")
+
+def pytest_generate_tests(metafunc):
+    """default is to test all non-sage kernels listed with `jupyter kernelspec list`"""
+    option_value = metafunc.config.option.kname
+    if 'kname' in metafunc.fixturenames and option_value is not None:
+        knames = option_value.split(',')
+        metafunc.parametrize("kname", knames)
+    # nsk = list of available non-sage kernel names
+    # skip first line of command output, "Available kernels"
+    else:
+        v = map(str.strip, os.popen("jupyter kernelspec list").readlines())
+        nsk = [x.split()[0] for x in v[1:] if not x.startswith('sage')]
```

Comment:
This `v[1:]` isn't python 3 ready, is it? change the `map( ... )` to comprehension like `[ x.strip() for x in ... ]`

```
$ python3
>>> x = [1,2,3]
>>> v = map(lambda z : z+22, x)
>>> v
<map object at 0x7f8bdd94f0b8>
>>> v[1:]
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: 'map' object is not subscriptable
```